From ed6c5cfbbbf021e36d9ee8c9760869c2589d2c29 Mon Sep 17 00:00:00 2001
From: ZhongWen <b47623@freescale.com>
Date: Thu, 24 Oct 2013 09:54:21 +0800
Subject: [PATCH 24/39] ENGR00281805 Dont show system on screen after reboot

Add public static final String SHOW_SYSTEM_TIME
and modify BootReceiver to start LoadSystemTime.class

Signed-off-by: ZhongWen <b47623@freescale.com>
---
 core/java/android/provider/Settings.java                           | 3 +++
 .../src/com/android/providers/settings/DatabaseHelper.java         | 1 +
 packages/SystemUI/src/com/android/systemui/BootReceiver.java       | 7 +++++++
 3 files changed, 11 insertions(+)

diff --git a/core/java/android/provider/Settings.java b/core/java/android/provider/Settings.java
index 7f24539..91d177a 100644
--- a/core/java/android/provider/Settings.java
+++ b/core/java/android/provider/Settings.java
@@ -1092,6 +1092,7 @@ public final class Settings {
             MOVED_TO_GLOBAL.add(Settings.Global.DEBUG_APP);
             MOVED_TO_GLOBAL.add(Settings.Global.WAIT_FOR_DEBUGGER);
             MOVED_TO_GLOBAL.add(Settings.Global.SHOW_PROCESSES);
+            MOVED_TO_GLOBAL.add(Settings.Global.SHOW_SYSTEM_TIME);
             MOVED_TO_GLOBAL.add(Settings.Global.ALWAYS_FINISH_ACTIVITIES);
             MOVED_TO_GLOBAL.add(Settings.Global.TZINFO_UPDATE_CONTENT_URL);
             MOVED_TO_GLOBAL.add(Settings.Global.TZINFO_UPDATE_METADATA_URL);
@@ -1841,6 +1842,7 @@ public final class Settings {
          */
         @Deprecated
         public static final String SHOW_PROCESSES = Global.SHOW_PROCESSES;
+        public static final String SHOW_SYSTEM_TIME = Global.SHOW_SYSTEM_TIME;
 
         /**
          * If 1, the activity manager will aggressively finish activities and
@@ -5820,6 +5822,7 @@ public final class Settings {
          * Control whether the process CPU usage meter should be shown.
          */
         public static final String SHOW_PROCESSES = "show_processes";
+        public static final String SHOW_SYSTEM_TIME = "show_system_time";
 
         /**
          * If 1, the activity manager will aggressively finish activities and
diff --git a/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java b/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
index d3a0850..39c7fbd 100644
--- a/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
+++ b/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
@@ -1427,6 +1427,7 @@ public class DatabaseHelper extends SQLiteOpenHelper {
                             Settings.Global.DEBUG_APP,
                             Settings.Global.WAIT_FOR_DEBUGGER,
                             Settings.Global.SHOW_PROCESSES,
+                            Settings.Global.SHOW_SYSTEM_TIME,
                             Settings.Global.ALWAYS_FINISH_ACTIVITIES,
                     };
                     String[] secureToGlobal = {
diff --git a/packages/SystemUI/src/com/android/systemui/BootReceiver.java b/packages/SystemUI/src/com/android/systemui/BootReceiver.java
index 8e24eeb..e253edf 100644
--- a/packages/SystemUI/src/com/android/systemui/BootReceiver.java
+++ b/packages/SystemUI/src/com/android/systemui/BootReceiver.java
@@ -21,6 +21,7 @@ import android.content.ContentResolver;
 import android.content.Context;
 import android.content.Intent;
 import android.provider.Settings;
+import android.util.Slog;
 import android.util.Log;
 
 /**
@@ -39,6 +40,12 @@ public class BootReceiver extends BroadcastReceiver {
                 Intent loadavg = new Intent(context, com.android.systemui.LoadAverageService.class);
                 context.startService(loadavg);
             }
+
+            // ContentResolver sys = context.getContentResolver();
+            if (Settings.Global.getInt(res,Settings.Global.SHOW_SYSTEM_TIME,0) != 0){
+                Intent loadsystime = new Intent(context, com.android.systemui.LoadSystemTime.class);
+                context.startService(loadsystime);
+            }
         } catch (Exception e) {
             Log.e(TAG, "Can't start load average service", e);
         }
-- 
1.8.0

